### ISSUE TITLE
wallabag share crashes

### ISSUE THREAD

**Issue by tcitworld** [jangernert/FeedReader#278]
When you want to add a new wallabag share, when clicking the login button (with or without values in the fields), you get a segfault.

Version : latest master, on debian unstable.

Relevant logs : 
```
[ DEBUG ] WallabagAPI getAccessToken

(feedreader:23864): libsoup-CRITICAL **: soup_message_set_request: assertion 'SOUP_IS_MESSAGE (msg)' failed

(feedreader:23864): libsoup-CRITICAL **: soup_session_send_message: assertion 'SOUP_IS_MESSAGE (msg)' failed
Erreur de segmentation
```

Please tell me if and how I can provide more informations.

**Comment by jangernert** [comment]
A wallabag instance where I can test the current code would be a nice start :)
v2.wallabag.org seems to be offline

**Comment by tcitworld** [comment]
There's a 14 days free subscription on wallabag.it, you can contact me if more is needed. :)

**Comment by jangernert** [comment]
Just tested it, and works perfectly =/
Anything special about your setup? Did you compile the source yourself or installed a package?
The code that is failing are these 3 lines:
```
var message_soup = new Soup.Message("POST", url);
message_soup.set_request("application/x-www-form-urlencoded; charset=UTF8", Soup.MemoryUse.COPY, message.data);
session.send_message(message_soup);
```
Somehow the object 'message_soup' is not a valid Soup.Message on your system.

**Comment by tcitworld** [comment]
>  Anything special about your setup? Did you compile the source yourself or installed a package?

Compiled myself. The libsoup version is `2.56.0-1`.
Recompiled without any changes.
Didn't find any more broke stuff.

**Comment by jangernert** [comment]
Hm, that is super weird. No idea what could go wrong with these 3 simple lines.
Can you get a backtrace of the crash with gdb?

**Comment by tcitworld** [comment]
Doesn't gives much informations : 
<details>
<code>

#0  0x00007fff6cfed8cc in feed_reader_wallabag_api_getAccessToken ()
   from /usr/local/share/FeedReader/pluginsShare//libwallabag.so
No symbol table info available.

#1  0x00007fff6cff3083 in feed_reader_wallabag_setup_real_login ()
   from /usr/local/share/FeedReader/pluginsShare//libwallabag.so
No symbol table info available.

#2  0x000055555562623d in feed_reader_service_setup_login ()
No symbol table info available.

#3  0x0000555555624ed8 in _feed_reader_service_setup_login_gtk_button_clicked
    ()
No symbol table info available.

#4  0x00007ffff03051a4 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#5  0x00007ffff031f8bd in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#6  0x00007ffff031ffaf in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#7  0x00007ffff3d0dc1d in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#8  0x00007ffff3d0dc85 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#9  0x00007ffff0304f75 in g_closure_invoke ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#10 0x00007ffff031695c in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#11 0x00007ffff031fbcc in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#12 0x00007ffff031ffaf in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#13 0x00007ffff3d0c080 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#14 0x00007fffe5c71038 in ffi_call_unix64 ()
   from /usr/lib/x86_64-linux-gnu/libffi.so.6
No symbol table info available.

#15 0x00007fffe5c70a9a in ffi_call ()
   from /usr/lib/x86_64-linux-gnu/libffi.so.6
No symbol table info available.

#16 0x00007ffff0305c8a in g_cclosure_marshal_generic_va ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#17 0x00007ffff03051a4 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#18 0x00007ffff031f8bd in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#19 0x00007ffff031ffaf in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#20 0x00007ffff3dc78f1 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#21 0x00007ffff0307d27 in g_cclosure_marshal_VOID__BOXEDv ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#22 0x00007ffff03051a4 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#23 0x00007ffff031f8bd in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#24 0x00007ffff031ffaf in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#25 0x00007ffff3dc4bbe in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#26 0x00007ffff3dc61eb in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#27 0x00007ffff3dc8e6d in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#28 0x00007ffff3d9630e in gtk_event_controller_handle_event ()
   from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#29 0x00007ffff3f5ae1b in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#30 0x00007ffff3e10971 in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#31 0x00007ffff03051a4 in ?? ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#32 0x00007ffff031f391 in g_signal_emit_valist ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#33 0x00007ffff031ffaf in g_signal_emit ()
   from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.

#34 0x00007ffff3f5d0bc in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#35 0x00007ffff3e0d92e in ?? () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#36 0x00007ffff3e0f9be in gtk_main_do_event ()
   from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.

#37 0x00007ffff3920bf5 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.

#38 0x00007ffff3951ba2 in ?? () from /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
No symbol table info available.

#39 0x00007ffff002b7f7 in g_main_context_dispatch ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#40 0x00007ffff002ba60 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.

#41 0x00007ffff002bb0c in g_main_context_iteration ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.

#42 0x00007ffff05e770d in g_application_run ()
   from /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
No symbol table info available.

#43 0x000055555558c8f7 in feed_reader_main ()
No symbol table info available.

#44 0x000055555558c942 in main ()
No symbol table info available.

</code>
</details>

**Comment by bil-elmoussaoui** [comment]
I'm able to reproduce this or maybe this issue that provoke this one too. 
Steps to reproduce 
1. Go and add a wallbag account, click on all the Gtk.Entry and then click on connect( don't know why it's not reporducable when you don't click on the Gtk.Entry)
gdb backtrace 
```
#0  0x00007fff661f274d in feed_reader_wallabag_api_getAccessToken ()
   from /usr/share/FeedReader/pluginsShare//libwallabag.so
#1  0x00007fff661f3b99 in ?? () from /usr/share/FeedReader/pluginsShare//libwallabag.so
#2  0x00007ffff2fdc334 in ?? () from /usr/lib/libgobject-2.0.so.0
#3  0x00007ffff2ff77dd in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
#4  0x00007ffff2ff7ecf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
#5  0x00007ffff48fec6d in ?? () from /usr/lib/libgtk-3.so.0
#6  0x00007ffff48fecd5 in ?? () from /usr/lib/libgtk-3.so.0
#7  0x00007ffff2fdc334 in ?? () from /usr/lib/libgobject-2.0.so.0
#8  0x00007ffff2ff77dd in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
#9  0x00007ffff2ff7ecf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
#10 0x00007ffff48fd0d0 in ?? () from /usr/lib/libgtk-3.so.0
#11 0x00007fffe87a71c8 in ffi_call_unix64 () from /usr/lib/libffi.so.6
---Type <return> to continue, or q <return> to quit---
#12 0x00007fffe87a6c2a in ffi_call () from /usr/lib/libffi.so.6
#13 0x00007ffff2fdce1a in g_cclosure_marshal_generic_va ()
   from /usr/lib/libgobject-2.0.so.0
#14 0x00007ffff2fdc334 in ?? () from /usr/lib/libgobject-2.0.so.0
#15 0x00007ffff2ff77dd in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
#16 0x00007ffff2ff7ecf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
#17 0x00007ffff49b8a01 in ?? () from /usr/lib/libgtk-3.so.0
#18 0x00007ffff2fdeeb7 in g_cclosure_marshal_VOID__BOXEDv ()
   from /usr/lib/libgobject-2.0.so.0
#19 0x00007ffff2fdc334 in ?? () from /usr/lib/libgobject-2.0.so.0
#20 0x00007ffff2ff77dd in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
#21 0x00007ffff2ff7ecf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
#22 0x00007ffff49b5cce in ?? () from /usr/lib/libgtk-3.so.0
---Type <return> to continue, or q <return> to quit---
#23 0x00007ffff49b72fb in ?? () from /usr/lib/libgtk-3.so.0
#24 0x00007ffff49b9f7d in ?? () from /usr/lib/libgtk-3.so.0
#25 0x00007ffff498740e in gtk_event_controller_handle_event ()
   from /usr/lib/libgtk-3.so.0
#26 0x00007ffff4b4c05b in ?? () from /usr/lib/libgtk-3.so.0
#27 0x00007ffff4a01a31 in ?? () from /usr/lib/libgtk-3.so.0
#28 0x00007ffff2fdc334 in ?? () from /usr/lib/libgobject-2.0.so.0
#29 0x00007ffff2ff72b1 in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
#30 0x00007ffff2ff7ecf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
#31 0x00007ffff4b4e2fc in ?? () from /usr/lib/libgtk-3.so.0
#32 0x00007ffff49fea3e in ?? () from /usr/lib/libgtk-3.so.0
#33 0x00007ffff4a00a7e in gtk_main_do_event () from /usr/lib/libgtk-3.so.0
#34 0x00007ffff4512be5 in ?? () from /usr/lib/libgdk-3.so.0
---Type <return> to continue, or q <return> to quit---
#35 0x00007ffff456e5d2 in ?? () from /usr/lib/libgdk-3.so.0
#36 0x00007ffff2d03687 in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
#37 0x00007ffff2d038f0 in ?? () from /usr/lib/libglib-2.0.so.0
#38 0x00007ffff2d0399c in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0
#39 0x00007ffff32c4fdd in g_application_run () from /usr/lib/libgio-2.0.so.0
#40 0x000000000042f07a in feed_reader_main ()
#41 0x00007ffff2035291 in __libc_start_main () from /usr/lib/libc.so.6
#42 0x000000000042558a in _start ()
```

**Comment by bil-elmoussaoui** [comment]
The message i get without gdb 
```

(feedreader:525): libsoup-CRITICAL **: soup_message_set_request: assertion 'SOUP_IS_MESSAGE (msg)' failed

(feedreader:525): libsoup-CRITICAL **: soup_session_send_message: assertion 'SOUP_IS_MESSAGE (msg)' failed

Thread 1 "feedreader" received signal SIGSEGV, Segmentation fault.
0x00007fff661f274d in feed_reader_wallabag_api_getAccessToken ()
   from /usr/share/FeedReader/pluginsShare//libwallabag.so
```

**Comment by jangernert** [comment]
What Gtk.Entry do I have to click before logging in?
I just created a new account on wallabag.it, created a new client in the developer section, named it FeedReader with the the callback url "feedreader://wallabag", then went back to feedreader, added a new wallabag account, put in the url (https://app.wallabag.it/), copied over client ID and secret, entered my username and password and THEN clicked login. Anything I missed?

**Comment by bil-elmoussaoui** [comment]
I don't enter anything, i keep everything empty. Just click on all of them one after an other then click on the login button

**Comment by jangernert** [comment]
Okay, I check every entry now before logging in. That prevents the url from being NULL.
But like I wrote earlier

> However the original reporter mentioned he could get it to crash with all the right values in the entries.

**Comment by bil-elmoussaoui** [comment]
@jangernert I still can reproduce the crash if i enter random data on all the gtk.entry :(

**Comment by jangernert** [comment]
So much edge-cases :D 
Check for valid url is now also in place.

**Comment by bil-elmoussaoui** [comment]
It doesn't crash anymore. Btw, the message is shown only once, and if you click again it's won't be shown anymore (the error messages). Also i was able to log in using random values for all the entries and google.com for the url :p is that normal?

**Comment by jangernert** [comment]
Login-issue is fixed. But the issue with the InfoBar is beyond me at the moment. It appears again if you focus another window. Maybe you can take a look at the code. It's the same issue with all the plugins
https://github.com/jangernert/FeedReader/blob/master/plugins/share/Wallabag/WallabagSetup.vala#L47

**Comment by bil-elmoussaoui** [comment]
I think i know what could cause that! i will fix it tomorrow evening ;)

**Comment by tcitworld** [comment]
The crash issue is indeed fixed. Can't get to login though, I'll try to see why later.

**Comment by jangernert** [comment]
My first guess would be your url is somehow invalid. That would explain why it crashed the soup.message in the first place ;)

**Comment by jangernert** [comment]
If you still can't get it to work pls open another issue :)

**Comment by tcitworld** [comment]
Done at https://github.com/jangernert/FeedReader/issues/326
